### PR TITLE
Fix pull-request command for fetch only remotes.

### PR DESCRIPTION
**Pull request by madhusudancs**
There are no tests in this PR yet. I am working on them. But I wanted to get the code review going and see if any of you have any opinions about this change. Please take a look.

Fixes one part of issue #876.

### CODE REVIEW

File: git/git.go [hunk 1]
```diff
@@ -163,8 +163,24 @@ func Log(sha1, sha2 string) (string, error) {
...
+		return nil, fmt.Errorf("couldn't list remote names: %v", err)
+	}
+
+	remotes := make(map[string]string)
+	for _, name := range names {
+		out, err := gitOutput("ls-remote", "--get-url", name)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get url for remote %s: %v", name, err)
+		}
+		if len(out) != 1 {
```
Comment (mislav) [inline git/git.go]: What is this check for?

File: github/remote.go [hunk 2]
```diff
@@ -27,25 +25,12 @@ func (remote *Remote) Project() (*Project, error) {
 }
 
 func Remotes() (remotes []Remote, err error) {
-	re := regexp.MustCompile(`(.+)\s+(.+)\s+\((push|fetch)\)`)
-
-	rs, err := git.Remotes()
+	remotesMap, err := git.Remotes()
```
Comment (mislav) [inline github/remote.go]: Nice simplification 👍

**Comment by mislav** [comment]
Thanks for starting this conversation! Few notes:

1. What is the earliest version where `ls-remote --get-url` is available?
2. Is there a way to read `pushUrl` as well, but fallback to `url` if the former is not a valid URL?

File: git/git.go (hunk 1 above)
Comment (madhusudancs) [inline git/git.go]: Sorry, I wanted to add an error message for the return which would have made the intention here clear, but I got distracted.

This is an assertion or a defense to ensure that git-ls-remote --get-url always returns one and only one URL, in the unlikely event that this changes.

**Comment by madhusudancs** [comment]
> 1. What is the earliest version where ls-remote --get-url is available?

It looks like it was added in 1.7.5 -https://github.com/git/git/commit/45781adb9a89c0c47f61ccf8062be26b86a38a54 but the documentation did not appear until 1.7.12.2 -https://github.com/git/git/commit/2303cad242a6ccc4745d7994ecc0425bd8513a0d

> 2. Is there a way to read pushUrl as well, but fallback to url if the former is not a valid URL?

I don't think it is possible to do that with ls-remote --get-url, unless there is a trick that I am unaware of. But if we want to do that we might have to end up parsing the output that git remote produces.

**Comment by mislav** [comment]
Thanks for the detective work. We don't have a defined requirement for minimum git version yet, but I'm worried about requiring 1.7.5 because that might be too high, and the payoff isn't really big.

Could you instead take a stab at a separate PR that keeps reading `git remote -v`, but ignores `pushUrl`s that are invalid URLs (some people use an invalid `pushUrl` to prevent accidental pushes)?

**Comment by madhusudancs** [comment]
> Could you instead take a stab at a separate PR that keeps reading git remote -v, but ignores pushUrls that are invalid URLs (some people use an invalid pushUrl to prevent accidental pushes)?

Sure.

**Comment by mislav** [comment]
Closing in favor of your new PR, thanks!

**Pull request closed without merging**